lib/sender.js

```diff
@@ -80,12 +83,27 @@ class Sender {
       }
 
       skipMasking = (mask[0] | mask[1] | mask[2] | mask[3]) === 0;
-      if (options.readOnly && !skipMasking) merge = true;
-
       offset = 6;
     }
 
-    let payloadLength = data.length;
+    let dataLength;
+
+    if (typeof data === 'string') {
+      if (!options.mask || skipMasking) {
+        dataLength =
+          options.byteLength !== undefined
```

Comment:
I wonder if it is better to hide this behind a symbol property (`options[kByteLength]`) so that the API cannot be misused. I'm don't know if symbol properties are still slower.